### PR TITLE
Of branch

### PR DESCRIPTION
**Pull request by Ohadforman**

### CODE REVIEW

File: Component_section/component_api.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        return None
+    else:
+        error_message = f"Error {response.status_code}: Unexpected error."
+        pprint(error_message)
+        return None
+
+def extract_prices(prices_dict):
+    # Adjust this function to match your actual data structure
+    return ", ".join([f"{currency}: {p['unit_price']} for {p['unit_break']} units" for currency, price_list in prices_dict.items() for p in price_list])
+
```
Comment (shaizae10) [inline Component_section/component_api.py:35]: add if name=="main

File: Utils/openai_integration.py [hunk 2]
```diff
@@ -1,19 +1,25 @@
...
+import os
 
 # OpenAIElectronicDesignAssistant Class
 class OpenAiApi:
     def __init__(self, config):
         self.model = config['MODEL']
-        self.api_key = config["API_KEY"]
+        
+        # Read API key from a file
+        api_key_path = os.path.join(os.path.dirname(__file__), "..", "keys", "openai_api_key.txt")
```
Comment (shaizae10) [inline Utils/openai_integration.py:10]: make privet
maby whit one why property

File: Component_section/component_api.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    if country_code:
+        params['countryCode'] = country_code
+    if currencies:
+        params['currency[]'] = ','.join(currencies)
+
+    response = requests.get(base_url, params=params)
+    if response.status_code == 200:
+        return response.json()
+    elif response.status_code in [400, 401, 404]:
+        error_message = f"Error {response.status_code}: {response.json().get('error_message', 'No error message')}"
```
Comment (shaizae10) [inline Component_section/component_api.py:24]: use raise error

File: Component_section/component_api.py (hunk 1 above)
Comment (Ohadforman) [reply Component_section/component_api.py]: done

File: Utils/openai_integration.py (hunk 2 above)
Comment (Ohadforman) [reply Utils/openai_integration.py]: done(I think)

File: Component_section/component_api.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+import pandas as pd
+from requests.exceptions import HTTPError
+
+
+def get_api_key(file_path):
+    with open(file_path, 'r') as file:
+        return file.readline().strip()
+
+
+def fetch_part_data(part_number, api_key, country_code=None, currencies=None):
```
Comment (Ohadforman) [inline Component_section/component_api.py:11]: it say api_key is shadow can you check?

**Review by Ohadforman — COMMENTED** [review]
shdow

File: Component_section/component_api.py (hunk 4 above)
Comment (Ohadforman) [reply Component_section/component_api.py:11]: @shaizae

**Pull request merged**